tensorflow-core/tensorflow-core-api/src/bazel/op_generator/op_specs.cc

```diff
@@ -391,8 +391,64 @@ OpSpec OpSpec::Create(const OpDef& op_def, const ApiDef& api_def) {
   for (const auto& endpoint_def : api_def.endpoint()) {
     op.endpoints_.push_back(CreateEndpoint(op_def, api_def, endpoint_def));
   }
+  op.remove_extra_generics();
   return op;
 }
 
+void OpSpec::remove_extra_generics() {
+  std::map<string, int> generics;
+
+  for (const ArgumentSpec& output : this->outputs()) {
+    if (output.type().kind() == Type::GENERIC && !output.type().wildcard()) {
+      if(generics.find(output.type().name()) == generics.end()) {
+        generics[output.type().name()] = 1;
+      } else {
+        generics[output.type().name()] = generics.find(output.type().name())->second + 1;
+      }
+    }
+  }
+
+  for (const ArgumentSpec& input : this->inputs()) {
+    if (input.type().kind() == Type::GENERIC && !input.type().wildcard()) {
+      if(generics.find(input.type().name()) == generics.end()) {
+        generics[input.type().name()] = 1;
+      } else {
+        generics[input.type().name()] = generics.find(input.type().name())->second + 1;
+      }
+    }
+  }
+
+
+  for (ArgumentSpec& output : this->outputs_) {
+    if (output.type().kind() == Type::GENERIC && !output.type().wildcard()) {
+      if(generics.find(output.type().name()) == generics.end() || generics[output.type().name()] <= 1) {
```

Comment:
it sounds to me that it is not possible that `generics.find(output.type().name()) == generics.end()` here, let's remove that condition to avoid confusion. Same thing below.